Gem/Code/Source/AutoGen/EnergyBallComponent.AutoComponent.xml

```diff
@@ -16,17 +16,18 @@
     <ArchetypeProperty Type="GameEffect" Name="ExplosionEffect" Init="" ExposeToEditor="true" Description="Specifies the effect to play upon exploding" />
     <ArchetypeProperty Type="GatherParams" Name="GatherParams" Init="" ExposeToEditor="true" Description="Specifies the types of intersections to test for on the projectile" />
     <ArchetypeProperty Type="HitEffect" Name="HitEffect" Init="" ExposeToEditor="true" Description="Specifies the damage effects to apply on hit" />
+    <ArchetypeProperty Type="AZ::TimeMs" Name="LifetimeMs" Init="AZ::TimeMs{ 0 }" Container="Object" ExposeToEditor="true" Description="Specifies the duration in milliseconds that the projectile should live for" />
 
-    <NetworkProperty Type="bool" Name="BallActive" Init="false" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="true" IsPredictable="false" ExposeToScript="false" ExposeToEditor="false" GenerateEventBindings="true" Description="Track whether or not the energy ball is currently active" />
+    <NetworkProperty Type="Multiplayer::NetEntityId" Name="ShooterNetEntityId" Init="Multiplayer::InvalidNetEntityId" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="true" IsPredictable="false" ExposeToScript="true" ExposeToEditor="false" GenerateEventBindings="true" Description="The net entityId of the entity that shot the energy ball" />
+    <NetworkProperty Type="AZ::Vector3" Name="Velocity" Init="AZ::Vector3::CreateZero()" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="true" IsPredictable="false" ExposeToScript="true" ExposeToEditor="false" GenerateEventBindings="true" Description="The energy balls current velocity" />
+    <NetworkProperty Type="bool" Name="BallActive" Init="false" ReplicateFrom="Authority" ReplicateTo="Client" Container="Object" IsPublic="true" IsRewindable="true" IsPredictable="false" ExposeToScript="true" ExposeToEditor="false" GenerateEventBindings="true" Description="Track whether or not the energy ball is currently active" />
```

Comment:
Yeah, turns out it was still necessary to play the explosion effect correctly. Since this is hooked up and working, we could use this to create a projectile pool fairly easily as well, so we're not constantly spawning and destroying projectile entities.